lib/vtls/rustls.c

```diff
@@ -161,20 +161,17 @@ cr_recv(struct Curl_easy *data, int sockindex,
       (uint8_t *)plainbuf + plain_bytes_copied,
       plainlen - plain_bytes_copied,
       &n);
-    if(rresult == RUSTLS_RESULT_ALERT_CLOSE_NOTIFY) {
+    if(n == 0) {
```

Comment:
We discussed this a little out-of-band, but writing it down here:

The check for `n == 0` has to come last. If the result was not `RUSTLS_RESULT_OK`, n was not written to.

The check for `RUSTLS_RESULT_PLAINTEXT_EMPTY` should come first. That way you don't have to repeat it in two `if` stanzas.

Reply:
OK, done - updated the code to match your new description.